conbench/api/compare.py

```diff
@@ -1,146 +1,273 @@
+            "improvement_indicated": improvement_indicated,
+        }
+
+    @property
+    def _dict_for_api_json(self) -> dict:
+        return {
+            "unit": self.unit,
+            "less_is_better": self.less_is_better,
+            "baseline": self.result_info(self.baseline),
+            "contender": self.result_info(self.contender),
+            "analysis": {
+                "pairwise": self.pairwise_analysis,
+                "lookback_z_score": self.lookback_z_score_analysis,
+            },
+        }
+
+
+class CompareBenchmarkResultsAPI(ApiEndpoint):
+    @staticmethod
+    def _get_a_result(benchmark_result_id: str) -> BenchmarkResult:
+        """Get a benchmark result by ID, or abort if it doesn't exist."""
+        benchmark_result = BenchmarkResult.get(benchmark_result_id)
+        if not benchmark_result:
+            f.abort(
+                404,
+                description=f"no benchmark result found with ID: '{benchmark_result_id}'",
+            )
+        return benchmark_result
 
     @maybe_login_required
-    def get(self, compare_ids) -> f.Response:
+    def get(self, compare_ids: str) -> f.Response:
         """
         ---
-        description: Compare benchmark results.
+        description: |
+            Compare benchmark results.
+
+            Returns basic information about the baseline and contender benchmark results
+            as well as some analyses comparing the performance of the contender to the
+            baseline.
+
+            The `pairwise` analysis computes the percentage difference of the
+            contender's mean value to the baseline's mean value. The reported difference
+            is signed such that a more negative value indicates more of a performance
+            regression. This difference is then thresholded such that values more
+            extreme than the threshold are marked as `regression_indicated` or
+            `improvement_indicated`. The threshold is 5.0% by default, but can be
+            changed via the `threshold` query parameter, which should be a positive
+            number representing a percent.
+
+            The `pairwise` analysis may be `null` if either benchmark result does not
+            have a mean value, or if the baseline result's mean value is 0.
+
+            The `lookback_z_score` analysis computes the z-score of the contender's mean
+            value compared to a distribution of benchmark results ending with the
+            baseline result. The reported z-score is also signed such that a more
+            negative value indicates more of a performance regression, and thresholded.
+            The threshold z-score is 5.0, but can be changed via the `threshold_z` query
+            parameter, which should be a positive number.
+
+            The `lookback_z_score` may be `null` if the contender benchmark result is
+            missing a z-score, due to not finding a baseline distribution that matches
```

Comment:
> The `lookback_z_score` may be `null` if the contender benchmark result is   missing a z-score

Is that like saying "the cow might be black if the cow is black"? :)

Reply:
Oops, forgot a few words!